### PR TITLE
closed streams detected by FindBugs

### PR DESCRIPTION
**Pull request by akmorrow13**

### CODE REVIEW

File: src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java [hunk 1]
```diff
@@ -328,139 +328,145 @@ private int addBAISplits(List<InputSplit> splits,
...
+						FileSplit fSplit = (FileSplit) splits.get(splitsEnd - 1);
+						long fSplitEnd = (fSplit.getStart() + fSplit.getLength()) << 16;
+						newSplit = new FileVirtualSplit(fSplit.getPath(),
+								lastStart,
+								fSplitEnd,
+								fSplit.getLocations());
+						newSplits.add(newSplit);
+					}
+					logger.info("closing");
+					System.out.println("sin and guesser closing00");
```
Comment (fnothaft) [inline src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java]: Please remove the debug printing.

File: src/main/java/org/seqdoop/hadoop_bam/util/BGZFBlockIndexer.java [hunk 2]
```diff
@@ -95,36 +89,36 @@ public BGZFBlockIndexer(int g) {
...
 				}
-			}
-		}
-		lb.put(0, file.length());
-		out.write(byteBuffer.array(), 2, 6);
-		out.close();
-		in.close();
+				lb.put(0, file.length());
+				out.write(byteBuffer.array(), 2, 6);
+			} catch (IOException e) {}
```
Comment (fnothaft) [inline src/main/java/org/seqdoop/hadoop_bam/util/BGZFBlockIndexer.java]: I wouldn't silently catch these exceptions.

File: src/main/java/org/seqdoop/hadoop_bam/util/GetSortedBAMHeader.java [hunk 3]
```diff
@@ -51,7 +51,9 @@ public static void main(String[] args) throws IOException {
 						.open(new File(args[0])).getFileHeader();
 		h.setSortOrder(SAMFileHeader.SortOrder.coordinate);
 
-		new SAMOutputPreparer().prepareForRecords(
-			new FileOutputStream(args[1]), SAMFormat.BAM, h);
+		try (FileOutputStream stream = new FileOutputStream(args[1])) {
+			new SAMOutputPreparer().prepareForRecords(
+					stream, SAMFormat.BAM, h);
+		} catch (IOException e) {}
```
Comment (fnothaft) [inline src/main/java/org/seqdoop/hadoop_bam/util/GetSortedBAMHeader.java]: Ditto here.

File: src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java [hunk 4]
```diff
@@ -328,139 +328,145 @@ private int addBAISplits(List<InputSplit> splits,
...
-                                FileSplit fSplit = (FileSplit)splits.get(splitsEnd - 1);
-                                long fSplitEnd = (fSplit.getStart() + fSplit.getLength()) << 16;
-                                newSplit = new FileVirtualSplit(fSplit.getPath(),
-                                                                lastStart,
-                                                                fSplitEnd,
-                                                                fSplit.getLocations());
-                                newSplits.add(newSplit);
-                        }
-                }
+					SAMFileHeader header = SAMHeaderReader.readSAMHeaderFrom(in, conf);
```
Comment (fnothaft) [inline src/main/java/org/seqdoop/hadoop_bam/BAMInputFormat.java]: Can we eliminate the indentation changes in this file?

**Comment by fnothaft** [comment]
Ping for review @tomwhite.

**Comment by tomwhite** [comment]
Ah, just saw this after merging #193. let me know if there's anything more needed or if this can be closed. Thanks.

**Pull request closed without merging**